.github/workflows/update-wasm-tests.yml

```diff
@@ -0,0 +1,65 @@
+name: Update WASM tests
+
+on:
+  # Trigger at every Sunday UTC noon, or manually.
+  schedule:
+    - cron: 0 12 * * 0
+  workflow_dispatch:
+
+jobs:
+  build-wpt:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout WPT repo
+        uses: actions/checkout@v4
+        with:
+          path: main
+      - name: Checkout WASM repo
+        uses: actions/checkout@v4
+        with:
+          repository: WebAssembly/spec
+          path: wasm-spec
+      - name: Setup OCaml
+        uses: ocaml/setup-ocaml@v3
+        with:
+          ocaml-compiler: 4.14.x
+      - name: Setup OCaml tools
+        run: opam install --yes ocamlfind.1.9.5 js_of_ocaml.4.0.0 js_of_ocaml-ppx.4.0.0
+      - name: Build interpreter
+        run: cd wasm-spec/interpreter && opam exec make
+      - name: Convert WAST tests to WPT
+        run: wasm-spec/test/build.py --dont-recompile --html wasm-spec/out/
+      - name: Copy WASM tests to WPT
+        # Replace wasm/core entirely and fix up testharness paths.
+        run: |
+          rm -rf main/wasm/core
+          cp -r wasm-spec/out/ main/wasm/core/
+          find main/wasm/core/ -type f -name '*.html' -exec sed -i 's/\.\/js\/harness\/testharness/\/resources\/testharness/' {} \;
+      - name: Commit changes
```

Comment:
If you use the `peter-evans/create-pull-request` action as the WebIDL workflow does, you don't need to manually commit the changes or do anything with git itself.

Reply:
Another example is what we do in [emscripten](https://github.com/emscripten-core/emscripten/blob/main/.github/workflows/tag-release.yml#L20) to create a PR, request a review from a github team, and set the PR to auto-merge (after approval and passing tests)

Reply:
We do something similar in the [wpt-metadata](https://github.com/web-platform-tests/wpt-metadata/tree/master/.github/workflows) repo, but I'm always wary of using 3rd-party libraries for security reasons. However, I see the author of this action is now working for GitHub, so maybe this is less of a concern.

Reply:
Yeah, that makes sense. I'm fine either way, up to you.